datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -1040,7 +1040,7 @@ mod test {
         let empty = empty_with_type(DataType::Int64);
         let plan = LogicalPlan::Projection(Projection::try_new(vec![expr], empty)?);
         let expected =
-            "Projection: a IN ([CAST(Int32(1) AS Int64), CAST(Int8(4) AS Int64), Int64(8)]) AS a IN (Map { iter: Iter([Int32(1), Int8(4), Int64(8)]) })\
+            "Projection: a IN ([CAST(Int32(1) AS Int64), CAST(Int8(4) AS Int64), Int64(8)]) AS a IN (Map { iter: Iter([Literal(Int32(1)), Literal(Int8(4)), Literal(Int64(8))]) })\
              \n  EmptyRelation";
```

Comment:
I think this test case should not change, but current I failed to past this test case.
When I change the code [0ef94](https://github.com/apache/arrow-datafusion/blob/0ef9478e6675308529d5420e94d70b0f3a32dc98/datafusion/expr/src/expr.rs#L1110) to use `Display` I got failed by `Projection: a IN ([CAST(Int32(1) AS Int64), CAST(Int8(4) AS Int64), Int64(8)])` missing `Alias`

Reply:
Which alias?

Reply:
` AS a IN (Map { iter: Iter([Int32(1), Int8(4), Int64(8)]) })\` is missing